File: app/src/java/java/com/longtoast/bilbil_api/dto/ProductCreateRequest.java
package com.longtoast.bilbil_api.dto;

import com.longtoast.bilbil_api.domain.Item;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

// 안드로이드 ProductCreateRequest.kt의 필드와 동일하게 맞춥니다.
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProductCreateRequest {

    private String title;
    private int price;
    private int price_unit;
    private String description;
    private String category;

    // 💡 [수정] Item.Status ENUM 타입으로 변경 (클라이언트에서는 String으로 전송)
    private Item.Status status;

    private Integer deposit;
    private String address;
}

File: app/src/main/java/com/longtoast/bilbil/ProductCreateRequest.kt
// com.longtoast.bilbil.dto/ProductCreateRequest.kt (전체)
package com.longtoast.bilbil.dto

// 서버에 게시글 작성을 위해 보낼 데이터를 정의합니다.
data class ProductCreateRequest(
    val title: String,
    val price: Int,
    val price_unit: Int,
    val description: String,
    val category: String,
    val status: String,
    val deposit: Int?,
    val imageUrls: List<String>,
    val address: String,
    val latitude: Double,      // 추가됨
    val longitude: Double      // 추가됨
)